cpp/src/arrow/compute/kernels/scalar_string_test.cc

```diff
@@ -68,6 +76,64 @@ TYPED_TEST(TestStringKernels, AsciiLower) {
                    this->string_type(), "[\"aaazzæÆ&\", null, \"\", \"bbb\"]");
 }
 
+TEST(TestStringKernels, Utf8Upper32bitGrowth) {
```

Comment:
I'm not convinced it's a good idea to consume more than 2GB RAM in a test. We have something called `LARGE_MEMORY_TEST` for such tests (you can grep for it), though I'm not sure they get exercised on a regular basis.

Reply:
I think it's fine to have this test but it definitely should be marked as LARGE_MEMORY_TEST

Reply:
Great, I was hoping such a system existed.